### PR TITLE
tutorial writeup for p2p vue demo

### PR DESCRIPTION
**Pull request by thisisjofrank**
adding p2p vue tutorial

### CODE REVIEW

File: content/tutorials/peer-to-peer-vue.textile [hunk 1]
```diff
@@ -0,0 +1,671 @@
...
+      for (let word of phrase) {
+        this.ably.sendMessage({ kind: "word", word: word, serverState: this.state });
+        await sleep(500);
+      }
+    }
+```
+
+"The full code sample has a much longer phrase embedded in it":https://github.com/ably/tutorials/blob/6f31c075fc6db42aeb7dda6128ee889503e79b06/p2p-vue-demo/p2p.lib.server.js#L15
+
+Here, we're taking a phrase, and sending it as a series of messages word-by-word, with a 500ms pause in between each word. We're using @async / await@ to pause our code, just be *resolving a promise* every 500ms.
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:503]: I think `just be` should be `just by`?

File: content/tutorials/peer-to-peer-vue.textile [hunk 2]
```diff
@@ -0,0 +1,671 @@
...
+
+Now we should update the client to listen for this message. Add a @case@ statement for the @word@ message to it's @onReceiveMessage@ handler.
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:505]: `message to it's` => `message to its`

File: content/tutorials/peer-to-peer-vue.textile [hunk 3]
```diff
@@ -0,0 +1,671 @@
...
+```[javascript]
+this.state = {
+  status: "disconnected",
+  receivedWords: ""
+  };
+```
+
+Next, we're going to make sure to display this in the UI of every attached client by binding the state property into the markup in @index.html@
+
+```html
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:536]: `html` => `[html]`

File: content/tutorials/peer-to-peer-vue.textile [hunk 4]
```diff
@@ -0,0 +1,671 @@
...
+```[javascript]
+computed: {
+  state: function() { ... },
+  transmittedServerState: function() { ... },
+  joinedOrHosting: function () { ... },
+  iAmHost: function() { return this.p2pServer != null; },
+}
+```
+
+So now, if you start a hosting a session, and click @Stream words to client@ you'll see a stream of words appear one by one in your UI.
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:581]: `So now, if you start a hosting a session` => `So now, if you start hosting a session`

File: content/tutorials/peer-to-peer-vue.textile [hunk 5]
```diff
@@ -0,0 +1,671 @@
...
+
+If your internet connection is disrupted, the Ably client, used by @P2PServer@, will buffer it's outbound messages until a connection can be re-established. Ably handles buffered connnections in a few different ways (documented here https://www.ably.io/documentation/realtime/connection)
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:583]: `will buffer it's outbound message`  => `will buffer its outbound message`

Comment (tomczoink) [reply content/tutorials/peer-to-peer-vue.textile:583]: `connnections` => `connections`

Comment (tomczoink) [reply content/tutorials/peer-to-peer-vue.textile:583]: Missing full stop at end of paragraph. 

Also as a personal thing I think a href for the link going something like `you can find more information in our "realtime connections documentation":/realtime/connection` works a bit better. If not, at least a link on the link (aka `"https://www.ably.io/documentation/realtime/connection":/realtime/connection`) would be good.

File: content/tutorials/peer-to-peer-vue.textile [hunk 6]
```diff
@@ -0,0 +1,671 @@
...
+
+bq.. The disconnected state is entered if an established connection is dropped, or if a connection attempt was unsuccessful. In the disconnected state the library will periodically attempt to open a new connection (approximately every 15 seconds), anticipating that the connection will be re-established soon and thus connection and channel continuity will be possible.
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:585]: I think `bq..` => `bq.`

File: content/tutorials/peer-to-peer-vue.textile [hunk 7]
```diff
@@ -0,0 +1,671 @@
...
+
+In this state, developers can continue to publish messages as they are automatically placed in a local queue, to be sent as soon as a connection is reestablished. Messages published by other clients whilst this client is disconnected will be delivered to it upon reconnection, so long as the connection was resumed within 2 minutes.
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:587]: `reestablished` => `re-established` to match other uses of the word

File: content/tutorials/peer-to-peer-vue.textile [hunk 8]
```diff
@@ -0,0 +1,671 @@
...
+
+After 2 minutes have elapsed, recovery is no longer possible and the connection will move to the suspended state.
+
+The suspended state is entered after a failed connection attempt if there has then been no connection for a period of two minutes. In the suspended state, the library will periodically attempt to open a new connection every 30 seconds. Developers are unable to publish messages in this state. A new connection attempt can also be triggered by an explicit call to @connect()@ on the Connection object.
+
+Once the connection has been re-established, channels will be automatically re-attached. The client has been disconnected for too long for them to resume from where they left off, so if it wants to catch up on messages published by other clients while it was disconnected, it needs to use the history API.
+
+p. So much like our @P2PServer@ will buffer outbound messages, any disconnected clients will receive all the messages sent in a window of up to **two minutes of disconnection**
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:595]: Missing fullstop at end.

File: content/tutorials/peer-to-peer-vue.textile [hunk 9]
```diff
@@ -0,0 +1,671 @@
...
+
+The "Rewind":https://www.ably.io/documentation/realtime/channels/channel-parameters/overview#supported-parameters parameter can be enabled when you subscribe to a channel, which will immediatey then receive up to the last 100 messages in the time window you specify.
+
+The caveat is that the client will be processing historical messages - so be sure you understand what that means for your specific application. In some scenarios, it may be more effective for another @peer@ or the @server@ to just send new clients the state they **need** rather than the client reconstructing it.
+
+To enable Rewind, we need to add an additional parameter to the @channels.get@ call, to add the parameter @ { params: { rewind: '1m' } }`. This tells our client to return up to the last 100 messages, from the minute. You can provide different time periods in here in either minutes ('2m') or seconds ('15s').
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:605]: `To enable Rewind, we need to add an additional parameter to the @channels.get@ call, to add the parameter @ { params: { rewind: '1m' } }` reads quite confusingly to me, maybe something like `To enable Rewind, we need to add the parameter @{ params: { rewind: '1m' } }@ to our @channel.get@ call.` ?

Comment (tomczoink) [reply content/tutorials/peer-to-peer-vue.textile:605]: The ` after the parameter example should be an @

Comment (tomczoink) [reply content/tutorials/peer-to-peer-vue.textile:605]: `from the minute` => `from the last minute` perhaps for clarity?

File: content/tutorials/peer-to-peer-vue.textile [hunk 10]
```diff
@@ -0,0 +1,671 @@
...
+  this.channel = await ably.channels.get(`p2p-sample-${uniqueId}`, { params: { rewind: '1m' } });  
+  ...
+}
+```
+
+You will now have a working peer to peer base on which you can build all kinds of apps and games!
+
+h3. But wait, there's more!
+
+This is a self contained demo of building peer to peer apps, hosting in browser tabs, using @Ably Channels@ as their communication medium.
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:620]: May be me mis-reading, but I think `building peer to peer apps, hosting in browser tabs` => `building peer to peer apps, hosted in browser tabs`

File: content/tutorials/peer-to-peer-vue.textile [hunk 11]
```diff
@@ -0,0 +1,671 @@
...
+
+We'd love to see what exciting apps or games you could build on top of this sort of pattern - and to that end, we made our ourselves!
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:622]: `our ourselves!` => `one ourselves!`

File: content/tutorials/peer-to-peer-vue.textile [hunk 12]
```diff
@@ -0,0 +1,671 @@
...
+
+We made a distributed bingo game called @Ablingo@ on top of this code sample - by adding additional messages, logic, and state to our server. You can read a detailed readme of how we extended this sample at the [repository here](https://github.com/thisisjofrank/Ablingo/blob/master/readme.md).
+
+
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:626]: Extra line

File: content/tutorials/peer-to-peer-vue.textile [hunk 13]
```diff
@@ -0,0 +1,671 @@
...
+You will now have a working peer to peer base on which you can build all kinds of apps and games!
+
+h3. But wait, there's more!
+
+This is a self contained demo of building peer to peer apps, hosting in browser tabs, using @Ably Channels@ as their communication medium.
+
+We'd love to see what exciting apps or games you could build on top of this sort of pattern - and to that end, we made our ourselves!
+
+We made a distributed bingo game called @Ablingo@ on top of this code sample - by adding additional messages, logic, and state to our server. You can read a detailed readme of how we extended this sample at the [repository here](https://github.com/thisisjofrank/Ablingo/blob/master/readme.md).
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:624]: `[repository here](https://github.com/thisisjofrank/Ablingo/blob/master/readme.md).` => `"repository here":https://github.com/thisisjofrank/Ablingo/blob/master/readme.md.`

File: content/tutorials/peer-to-peer-vue.textile [hunk 14]
```diff
@@ -0,0 +1,671 @@
...
+
+We need to keep the @Ably API key@ on the server side, so that people can't grab it and use up your personal quota. The client side SDK can request a temporary key from an API call, we just need somewhere to host it. In the @api@ directory, there's code for an @Azure Functions@ API that implements this @Token Authentication@ behaviour.
+
+Azure Static Web Apps automatically hosts this API for us. To have this same experience locally, we'll need to use the "Azure Functions Core Tools":https://docs.microsoft.com/en-us/learn/modules/develop-test-deploy-azure-functions-with-core-tools/.
+
+h3. Local dev pre-requirements
+
+We'll use [live-server](https://www.npmjs.com/package/live-server) to serve the static files and [Azure functions core tools](https://www.npmjs.com/package/azure-functions-core-tools) for interactivity. In your project folder run:
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:637]: Links ain't doin' the linkin'

File: content/tutorials/peer-to-peer-vue.textile [hunk 15]
```diff
@@ -0,0 +1,671 @@
...
+
+```
```
Comment (tomczoink) [inline content/tutorials/peer-to-peer-vue.textile:639]: \`\`\` => ```[sh], same goes for all below code snippets.

File: content/tutorials/peer-to-peer-vue.textile (hunk 10 above)
Comment (thisisjofrank) [reply content/tutorials/peer-to-peer-vue.textile:620]: yeah, its not super clear is it, I meant that its a demo of hosting a session in a browser tab

File: content/tutorials/peer-to-peer-vue.textile (hunk 15 above)
Comment (thisisjofrank) [reply content/tutorials/peer-to-peer-vue.textile:639]: ah!

**Comment by MarkWoulfeAbly** [comment]
@thisisjofrank ran the page through a spellcheck, as long as you're happy with the changes I've pushed LGTM.

**Pull request merged**